heron/api/src/java/com/twitter/heron/streamlet/impl/operators/ReduceByWindowOperator.java

```diff
@@ -49,14 +52,10 @@ public void prepare(Map map, TopologyContext topologyContext, OutputCollector ou
   @SuppressWarnings("unchecked")
   @Override
   public void execute(TupleWindow inputWindow) {
-    I reducedValue = null;
+    T reducedValue = identity;
```

Comment:
shouldn't a null check be conducted here? if identity is null, it should be set to to the first tuple or something